src/utils/arrow.rs

```diff
@@ -53,13 +102,64 @@ pub(crate) fn arrow_schema_from_column_names_and_range(
     range: &Range<CalDataType>,
     column_names: &[String],
     row_idx: usize,
+    row_limit: usize,
 ) -> Result<Schema> {
     let mut fields = Vec::with_capacity(column_names.len());
 
     for (col_idx, name) in column_names.iter().enumerate() {
-        let col_type = get_arrow_column_type(range, row_idx, col_idx)?;
+        let col_type = get_arrow_column_type(range, row_idx, row_limit, col_idx)?;
         fields.push(Field::new(&alias_for_name(name, &fields), col_type, true));
     }
 
     Ok(Schema::new(fields))
 }
+
+#[cfg(test)]
+mod tests {
+    use calamine::Cell;
+    use rstest::{fixture, rstest};
+
+    use super::*;
+
+    #[fixture]
+    fn range() -> Range<CalDataType> {
+        Range::from_sparse(vec![
+            // First column
+            Cell::new((0, 0), CalDataType::Bool(true)),
+            Cell::new((1, 0), CalDataType::Bool(false)),
+            Cell::new((2, 0), CalDataType::Int(42)),
+            Cell::new((3, 0), CalDataType::Float(13.37)),
+            Cell::new((4, 0), CalDataType::String("hello".to_string())),
+            Cell::new((5, 0), CalDataType::Empty),
+        ])
+    }
+
+    #[rstest]
+    // pure bool
+    #[case(0, 2, ArrowDataType::Boolean)]
+    // pure int
+    #[case(2, 3, ArrowDataType::Int64)]
+    // pure float
+    #[case(3, 4, ArrowDataType::Float64)]
+    // pure string
+    #[case(4, 5, ArrowDataType::Utf8)]
+    // pure int + float
+    #[case(2, 4, ArrowDataType::Float64)]
```

Comment:
Note for myself: add int + null

Reply:
[025bdc1](https://github.com/ToucanToco/fastexcel/pull/164/commits/025bdc1af50fbc153655ba6a67808c19ae66e5dd)